### PR TITLE
Refactor to merge macpro code into macfc action

### PR DESCRIPTION
**Pull request by andrew-corbalt**

### CODE REVIEW

File: src/index.ts
Comment (hsoule-corbalt) [inline src/index.ts:7]: Did you confirm that if `core.getInput(inputName)` cannot find an input for `inputName` it will return an empty string,  rather than `undefined`?

File: src/libs/jira-lib.ts
Comment (hsoule-corbalt) [inline src/libs/jira-lib.ts:319]: Should we log or throw an error here if `issueId` is empty or undefined?

File: src/libs/security-hub-lib.ts
Comment (hsoule-corbalt) [inline src/libs/security-hub-lib.ts]: Is this just checking if `process.env. SECURITY_HUB_NEW_ISSUE_DELAY` is set or not? If so can you just do `!process.env. SECURITY_HUB_NEW_ISSUE_DELAY ? ...` rather than use `typeof`?

File: src/libs/jira-lib.ts
Comment (hsoule-corbalt) [inline src/libs/jira-lib.ts]: Copy pasta from the `transitionIssue` function above? Shouldn't the error message be `Error retrieveing issue`?

File: src/libs/jira-lib.ts
Comment (hsoule-corbalt) [inline src/libs/jira-lib.ts:83]: Did you confirm that this delete request works with the `params` object? When I made my [PR](https://github.com/Enterprise-CMCS/macpro-security-hub-sync/blob/1205dd9501938029b18837795b3a0531ad3d6a3e/src/libs/jira-lib.ts#L84), I had to explicitly set `username` as a query parameter in the URL. Just want to be sure it still works with this syntax.

File: src/libs/jira-lib.ts
Comment (hsoule-corbalt) [inline src/libs/jira-lib.ts:257]: It looks like `queue` is always going to have a length of 1, so is there any reason why you can't just use `queue.pop()`, and remove the `while` loop? I think that may be a slightly more straightforward way of doing things?

File: src/libs/jira-lib.ts
Comment (andrew-corbalt) [reply src/libs/jira-lib.ts:83]: Hmm, not sure what issue you ran into but I verified this works fine at least against my local container jira.

File: src/libs/jira-lib.ts
Comment (andrew-corbalt) [reply src/libs/jira-lib.ts]: Yeah, copy pasta, however I realized this function just isn't used anywhere, so I've removed it. In the transition issue code they get the issue and set it to a variable but then never actually do anything with it. (Thank you eslint)

File: src/libs/security-hub-lib.ts
Comment (andrew-corbalt) [reply src/libs/security-hub-lib.ts]: I just copied this code over from the macpro code base. That does appear to be what they are doing. copilot suggests
const delayForNewIssues = +(process.env.SECURITY_HUB_NEW_ISSUE_DELAY ?? "86400000"); // 24 * 60 * 60 * 1000
const maxDatetime = new Date(currentTime.getTime() - delayForNewIssues);
Which looks good to me. The conversion from string to number creates some subtle problems which this handles reasonably well in 1 line of code.

File: src/libs/jira-lib.ts
Comment (andrew-corbalt) [reply src/libs/jira-lib.ts:319]: Hmm this looks to be logic pulled over from macpro. I don't seen any need for this to be here at all, why are you calling closed issue with an undefined input in the first place? 

Since we are planning on rewriting the workflow code, lets hold off on changes transition to close related code.

Comment (hsoule-corbalt) [reply src/libs/jira-lib.ts:319]: 👍

File: src/libs/jira-lib.ts
Comment (hsoule-corbalt) [reply src/libs/jira-lib.ts:83]: Also works against my local jira; seems like this syntax works just fine.

File: test-infrastructure/jira-container/README.md
Comment (hsoule-corbalt) [inline test-infrastructure/jira-container/README.md]: Nit: typo: 'intended to be **use**' -> 'intended to be **used**'

File: test-infrastructure/jira-container/README.md
Comment (hsoule-corbalt) [inline test-infrastructure/jira-container/README.md]: To get this to build successfully on my Mac I had to add `--platform linux/amd64`. Is it worth specifying this in the README?

File: src/libs/security-hub-lib.ts
Comment (hsoule-corbalt) [reply src/libs/security-hub-lib.ts]: Did you want to add this change in for this PR?

File: action.yml
Comment (hsoule-corbalt) [inline action.yml:15]: I know we said we were going to plan to version the action, so I think we'll need to ensure we're doing that correctly before we merge this PR. Because this update to the `jira-host` input will break this action for anyone currently using it.

**Review by hsoule-corbalt — CHANGES_REQUESTED** [review]
I tested this locally against the Jira Docker container and everything works. Just have a few more comments. Let me know if you want to discuss any of this further!

File: src/index.ts
Comment (andrew-corbalt) [reply src/index.ts:7]: from getInput docs: "Returns an empty string if the value is not defined."

File: test-infrastructure/jira-container/README.md
Comment (andrew-corbalt) [reply test-infrastructure/jira-container/README.md]: Updated

File: action.yml
Comment (andrew-corbalt) [reply action.yml:15]: I changed the target branch to a branch "v2" So we can have multiple PRs targeting that branch until all refactor tasks are complete then we can move v2 over to main once we are ready.

File: src/index.ts
Comment (hsoule-corbalt) [reply src/index.ts:7]: 👍

File: src/libs/jira-lib.ts
Comment (hsoule-corbalt) [reply src/libs/jira-lib.ts]: 👍

File: src/libs/security-hub-lib.ts
Comment (andrewcremins) [reply src/libs/security-hub-lib.ts]: Added

Comment (andrew-corbalt) [reply src/libs/security-hub-lib.ts]: added

**Review by hsoule-corbalt — APPROVED** [review]
LGTM!

**Pull request merged**
